Assets/Mirror/Runtime/ProfilerData.cs

```diff
@@ -0,0 +1,109 @@
+using System;
+using System.Diagnostics;
+using UnityEngine.Events;
+
+namespace Mirror
+{
+    /// <summary>
+    /// Provides profiling information from mirror
+    /// A profiler can subscribe to these events and
+    /// present the data in a friendly way to the user
+    /// </summary>
+    public static class ProfilerData
+    {
+        #region Out messages
+
+        /// <summary>
+        /// Describes an outgoing message
+        /// </summary>
+        public readonly struct OutMessage
```

Comment:
Why not make this a single object to make the API simpler and just set a Direction flag on the message instead?